src/block_cache.cpp

```diff
@@ -229,7 +229,7 @@ static_assert(int(job_action_name.size()) == static_cast<int>(job_action_t::num_
 	char const* job_name(job_action_t const job)
 	{
 		int const j = static_cast<int>(job);
-		if (j < 0 || j >= piece_log_t::last_job)
```

Comment:
I think the same thing applies here. there are ways for ``j`` to be negative. my understanding is that casting a large ``unsigned int`` into an ``int`` is not undefined, just implementation defined

Reply:
auto const j = static_cast<std::uint8_t>(job);
?

Reply:
I think the reasoning here should be the same as the one above, about ``operation_t``